src/accessory.ts

```diff
@@ -431,8 +431,13 @@ export default class RoombaAccessory implements AccessoryPlugin {
 
                     if (state.running) {
                         this.log("Roomba is pausing");
-
                         await roomba.pause();
+                        if (this.DockOnStop) {
```

Comment:
I didn't think we should wait for the pause to succeed before calling back to Homebridge. I'm not sure though. Perhaps we should merge the state and then callback, and _then_ actually send the command. 

Or maybe it's okay for setter callbacks to be called after some time. 

What are your thoughts around making this change?

Reply:
I'm fine to revert the order if you think that's best. The main reason for making this change was due to the responsiveness of the robot, the home app, and their synchronization. However, that could have more to do with the connections than anything. The differences are minimal.

As far as a best practice related to HomeKit and the cached status - that I'm not sure.